lib/lua/utils.lua

```diff
@@ -136,4 +136,39 @@ function util.table.get_random(tab)
     end
 end
 
-return util
+-- List of programs that should be spawned each start/restart.
+util.program = {}
+util.program.programs = {}
+
+-- Registers the program to spawn at startup and every time it restarts
+-- @param bin The program to run. Can be an absolute path or a command to run.
+-- @param args The arguments (as a string) to pass to the program.
+function util.program.spawn_at_startup(bin, args)
+  assert(type(bin) == 'string', 'Non string given for program')
+  table.insert(util.program.programs, {
+                 bin = bin,
+                 args = args
+  })
+end
+
+-- Spawns the startup programs
+function util.program.spawn_startup_programs()
+  for index, program in ipairs(util.program.programs) do
+    os.execute(program.bin .. " " .. program.args .. " &")
+  end
+end
+
+-- Stops the startup programs. Does not remove them from the global list.
+function util.program.terminate_startup_programs()
+  for index, program in ipairs(util.program.programs) do
+    -- TODO Store PIDs and kill those instead
```

Comment:
I thought about it a bit.
Everything specifically related to way-cooler (like way-cooler-bg and way-cooler-bar) should be *pkilled*.

Then close each window in all the workspaces, this helps since the cmd-line programs won't be killed.

This should be enough hopefully, rest of the cleanup can be done in the shutdown function.